src/services/collaborativesearch.service.ts

```diff
@@ -829,6 +878,168 @@ export class CollaborativesearchService {
         return result;
     }
 
+
+    private computeResolveWithAbort(projection: [projType.aggregate, Array<Aggregation>]
+        | [projType.search, Search]
+        | [projType.geoaggregate, Array<Aggregation>]
+        | [projType.geohashgeoaggregate, GeohashAggregation]
+        | [projType.geosearch, Search]
+        | [projType.tiledgeosearch, TiledSearch]
+        | [projType.count, Count]
+        | [projType.range, RangeRequest]
+        | [projType.compute, ComputationRequest], abortableSignal, filters: Array<Filter>, isFlat?: boolean, max_age = this.max_age
+    ): Observable<any> {
+        const finalFilter = this.getFinalFilter(filters);
+        const dateformat = finalFilter.dateformat;
+        let aggregationRequest: AggregationsRequest;
+        let aggregationsForGet: string[];
+        let includes: string[] = [];
+        let excludes: string[] = [];
+        let returned_geometries: string;
+        let result;
+        const fForGet = this.buildFilterFieldGetParam('f', finalFilter);
+        const qForGet = this.buildFilterFieldGetParam('q', finalFilter);
+        let search: Search;
+        let pretty = false;
+        let flat = false;
+        let pageAfter;
+        let pageBefore;
+        let pageFrom;
+        let pageSize;
+        let pageSort;
+        if (projection[0] === projType.search.valueOf() || projection[0] === projType.geosearch.valueOf() ||
+            projection[0] === projType.tiledgeosearch.valueOf()) {
+            search = projection[0] === projType.tiledgeosearch.valueOf() ? (<TiledSearch>projection[1]).search : (<Search>projection[1]);
+            includes = [];
+            excludes = [];
+            if (search.projection !== undefined) {
+                if (search.projection.excludes !== undefined) {
+                    excludes.push(search.projection.excludes);
+                }
+                if (search.projection.includes !== undefined) {
+                    includes.push(search.projection.includes);
+                }
+            }
+            if (search.returned_geometries !== undefined) {
```

Comment:
For each variable that you assign, do you really need to check `if (!== undefined)`?
I mean, in this case `returned_geometries` is by default null. Then if you do `                returned_geometries = search.returned_geometries;` without the undefined check, is there a difference?

Reply:
I agree. Fixed